src/ucs/config/parser.c

```diff
@@ -1175,25 +1175,38 @@ ucs_status_t ucs_config_parser_fill_opts(void *opts, ucs_config_field_t *fields,
                                          int ignore_errors)
 {
     ucs_status_t status;
-    char prefix[128];
+    int cstm_prefix_len;
+    int full_prefix_len;
 
     /* Set default values */
     status = ucs_config_parser_set_default_values(opts, fields);
+    ucs_assert(env_prefix != NULL);
     if (status != UCS_OK) {
         goto err;
     }
 
+    /* Find if env_prefix consists of base prefix and custom prefix, base prefix
+     * should not contain underscores "_" except at the very end
+     * e.g in env_prefix = "LONG_PREFIX_UCX_LOG_LEVEL":
```

Comment:
should it be env_prefix = "LONG_PREFIX_UCX_" here?
also, not sure LOG_LEVEL is good example, we don't support custom prefix from global options right?

Reply:
That's right